code.gs

```diff
@@ -414,114 +415,157 @@ Contact.prototype.getPlainTextLines = function (type, date) {
     if (type === 'CUSTOM') {
-      htmlLine.push('&lt;' + htmlEscape(event.getProp('label')) + '&gt; ');
+      switch (format) {
+        case 'plain':
+          line.push('<' + beautifyLabel(event.getProp('label')) + '> ');
+          break;
+        case 'html':
+          line.push('&lt;' + htmlEscape(beautifyLabel(event.getProp('label'))) + '&gt; ');
+      }
     }
     // Full name.
-    htmlLine.push(htmlEscape(self.data.getProp('fullName')));
+    switch (format) {
+      case 'plain':
+        line.push(self.data.getProp('fullName'));
+        break;
+      case 'html':
+        line.push(htmlEscape(self.data.getProp('fullName')));
+    }
     // Nickname.
     if (!self.data.isPropEmpty('nickname')) {
-      htmlLine.push(' &quot;', htmlEscape(self.data.getProp('nickname')), '&quot;');
+      switch (format) {
+        case 'plain':
+          line.push(' "', self.data.getProp('nickname'), '"');
+          break;
+        case 'html':
+          line.push(' &quot;', htmlEscape(self.data.getProp('nickname')), '&quot;');
+      }
     }
-    // Age/years passed.
+    // Age/years passed
     if (!event.isPropEmpty('year')) {
       if (type === 'BIRTHDAY') {
-        htmlLine.push(' - ', htmlEscape(_('Age')), ': ');
+        switch (format) {
+          case 'plain':
+            line.push(' - ', _('Age'), ': ');
+            break;
+          case 'html':
+            line.push(' - ', htmlEscape(_('Age')), ': ');
+        }
       } else {
-        htmlLine.push(' - ', htmlEscape(_('Years')), ': ');
+        switch (format) {
+          case 'plain':
+            line.push(' - ', _('Years'), ': ');
+            break;
+          case 'html':
+            line.push(' - ', htmlEscape(_('Years')), ': ');
+        }
       }
-      htmlLine.push(Math.round(date.getYear() - event.getProp('year')));
+      line.push(Math.round(date.getYear() - event.getProp('year')));
     }
     // Email addresses and phone numbers.
     if (self.emails.length + self.phones.length > 0) {
-      htmlLine.push(' (');
-      htmlLine.push(
-        self.emails.map(function (email) {
-          return (!email.isPropEmpty('label') ? '[' + htmlEscape(beautifyLabel(email.getProp('label'))) + '] ' : '') + '<a href="mailto:' +
-            email.getProp('address') + '">' + htmlEscape(email.getProp('address')) + '</a>';
-        }).concat(self.phones.map(function (phone) {
-          return (!phone.isPropEmpty('label') ? '[' + htmlEscape(beautifyLabel(phone.getProp('label'))) + '] ' : '') + '<a href="tel:' +
-          phone.getProp('number') + '">' + htmlEscape(phone.getProp('number')) + '</a>';
-        })).join(' - '));
-      htmlLine.push(')');
+      var output, other;
+
+      output = [];
+      other = [];
+      // Email addresses.
+      self.emails.forEach(function (email) {
+        var label, emailAddr;
+
+        label = email.getProp('label');
+        emailAddr = email.getProp('address');
+        switch (label) {
+          case 'HOME_EMAIL':
```

Comment:
I agree with @baatochan: having all the emails grouped by label, considering all the possible labels instead of just two would be better.

This is a sketch in javascript/pseudocode of what I'm thinking about:

```javascript
groupedEmails = {}
foreach label, email in emails
  groupedEmails[label].append(email)

emails = '(' + groupedEmails_labels.map(label -> beautifyLabel(label) + ': ' + groupedEmails[label].join(' - ')).join(', ') + ')'
```

In this case "Other emails" would only be those addresses without a label or with label OTHER (which we've discovered actually being an empty string, so is considered empty by DataCollectors).

The same would be applied to phone numbers.

Reply:
We don't have to group emails and phones by ALL labels (considering there is like 10 phone labels), but we could take like:
* home email
* work email
* other emails

* mobile phone
* home phone
* work phone
* other phones

I also started to think as ppl can use customlabels (which can be important to them) to do something like that:

> <img> First Name Last Name Info about event Yesrs and do on/n
> Emails: /n
> Home email: xxx@xxx.xx - yyy@yy.yy /n
> Work email: zzz@zzz.zz - aaa@aaa.aa /n
> Other emails: bbb@bb.bb - ccc@cc.cc (personal custom label) - dd@dd.dd /n
> /n
> Phones: /n
> Mobile phone: 0000000 - 1111111 /n
> Work phone: 22222 - 333333 /n
> Home phone: 55555555 /n
> Other phones: 343453546 - 2345435 (custom label) - 52354654 - 43254654 - 435345132 (other custom label) /n
> /n
> <next contact>